test/toxiproxy_test.rb

```diff
@@ -426,6 +426,25 @@ def test_invalid_direction
     end
   end
 
+  def test_toxiproxy_whitelists_webmock
+    WebMock.enable!
+    WebMock.disable_net_connect!
+    endpoint = "#{Toxiproxy.uri.host}:#{Toxiproxy.uri.port}"
+
+    WebMock::Config.instance.allow = nil
+    Toxiproxy.version # This should initialize the list.
+    assert WebMock::Config.instance.allow.include?(endpoint)
+
+    WebMock::Config.instance.allow = ['some-other-host']
+    Toxiproxy.version # This should not remove 'some-other-host'.
+    assert WebMock::Config.instance.allow.include?('some-other-host')
+    assert WebMock::Config.instance.allow.include?(endpoint)
+    Toxiproxy.version # This should not duplicate the endpoint.
+    assert_equal 1, WebMock::Config.instance.allow.count(endpoint)
```

Comment:
This block I think is a different test `test_toxiproxy_whitelists_with_webmock_after_override`.